### PR TITLE
Add configuration for process cache size

### LINKED ISSUE #24346: Allow configuring cache size of process cache

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Allow configuring `maxCacheSize` for the ProcessCache. 

## Related issues

closes #24346

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ConfigValidator.java [hunk 1]
```diff
@@ -111,5 +111,11 @@ public static void validate(final ExporterConfiguration configuration) {
       throw new ExporterException(
           "CamundaExporter delayBetweenRuns must be >= 1. Current value: " + delayBetweenRuns);
     }
+
+    final int maxCacheSize = configuration.getProcessCache().getMaxCacheSize();
+    if (maxCacheSize < 1) {
+      throw new ExporterException(
+          "CamundaExporter maxCacheSize must be >= 1. Current value: " + maxCacheSize);
```
Comment (Zelldon) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ConfigValidator.java:118]:
```suggestion
          "CamundaExporter processCache.maxCacheSize must be >= 1. Current value: " + maxCacheSize);
```

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ConfigValidator.java [hunk 2]
```diff
@@ -111,5 +111,11 @@ public static void validate(final ExporterConfiguration configuration) {
       throw new ExporterException(
           "CamundaExporter delayBetweenRuns must be >= 1. Current value: " + delayBetweenRuns);
     }
+
+    final int maxCacheSize = configuration.getProcessCache().getMaxCacheSize();
```
Comment (Zelldon) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ConfigValidator.java:115]: Should check whether getProcessCache is null

File: zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/config/ConfigValidatorTest.java [hunk 3]
```diff
@@ -137,4 +137,16 @@ void shouldForbidDelayBetweenRunsToBeLessThanOne() {
         .isInstanceOf(ExporterException.class)
         .hasMessageContaining("CamundaExporter delayBetweenRuns must be >= 1. Current value: 0");
   }
+
+  @ParameterizedTest(name = "{0}")
+  @ValueSource(ints = {-1, 0})
+  void shouldForbidNonPositiveMaxCacheSize(final int maxCacheSize) {
```
Comment (Zelldon) [inline zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/config/ConfigValidatorTest.java:143]: Add a test that succeeds

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @deepthidevaki 

as discussed I will take over the PR and make adjustments where I see the need

**Review by ChrisKujawa — APPROVED** [review]

**Pull request merged**